src/utils/extractHeaders.ts

```diff
@@ -54,7 +55,28 @@ function processHeaders(
     const headers = response.protocol.http?.headers;
     if (headers) {
       const headerSchema = headersToSchema(headers, operationName, isException);
-      headerSchema && responseHeaders.push(headerSchema);
+      if (headerSchema) {
+        let pushHeaderSchema: boolean = true;
+        for (let responseHeader of responseHeaders) {
+          if (
+            getLanguageMetadata(responseHeader.language).name ==
```

Comment:
I think this looks good. For readability, would you mind extracting this section into a separate function and add some comments please?

Reply:
Done